src/main/java/TiTi/util/Parser.java

```diff
@@ -0,0 +1,124 @@
+package TiTi.util;
+
+import TiTi.task.Task;
+import TiTi.task.Event;
+import TiTi.task.Deadline;
+import TiTi.task.ToDo;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Make sense of user command, and return Response
+ * to tell Ui how to interact with user.
+ */
+public class Parser {
+    private SavedHistory savedHistory;
+    private TaskList taskList;
+    private Scanner sc;
+
+    /**
+     * Constructor for Parser class.
+     *
+     * @param savedHistory loads and writes to the saved data
+     * @param taskList list of current tasks
+     */
+    public Parser(SavedHistory savedHistory, TaskList taskList) {
+        this.savedHistory = savedHistory;
+        this.taskList = taskList;
+        sc = new Scanner(System.in);
+    }
+
+    /**
+     * Ask and interpret user command, and update list as required.
+     *
+     * @return Response object to tell Ui how to respons to user command
+     */
+    public Response cue() {
+        String input = sc.nextLine();
+        String cue = input.split(" ", 2)[0];
+
+        // handle missing description
+        if ((cue.equals("todo") || cue.equals("deadline")
+                || cue.equals("event") || cue.equals("find"))
+                && input.split(" ", 2).length == 1) {
```

Comment:
This expression seems a little bit complicated. Perhaps you could set aside a boolean `isDescriptionMissing` to handle this checking.